crates/network-tunnel/src/sshforwarding.rs

```diff
@@ -235,6 +236,19 @@ impl NetworkTunnel for SshForwarding {
 
         Ok(())
     }
+
+    async fn cleanup(&mut self) -> Result<(), Error> {
+        if let Some(process) = self.process.as_mut() {
+            process.kill().await?;
```

Comment:
Just double-checking that I've understood the logic correctly: if the process has already exited [this will result in an InvalidInput error](https://doc.rust-lang.org/std/process/struct.Child.html#method.kill), but that's okay because the code in `run_and_cleanup` discards any cleanup errors and just returns the result status of the `prepare; start_serve` sequence. Does that sound correct?

Reply:
@willdonnelly `run_and_cleanup` doesn't discard errors of `cleanup`, since it's using `?`. I just updated the code for this cleanup method to avoid erroring in case of `InvalidInput`, thanks for bringing this up!